artifactory/reportusage/reportusage.go

```diff
@@ -0,0 +1,80 @@
+package reportusage
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/utils/version"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+const SystemUsageApi = "6.9.0"
+
+func SendReportUsage(productId, commandName string, serviceManager *artifactory.ArtifactoryServicesManager) error {
+	config := serviceManager.GetConfig()
+	if config == nil {
+		return errors.New("Expected full config, but no configuration were configured.")
+	}
+	rtDetails := config.GetArtDetails()
+	if rtDetails == nil {
+		return errors.New("Expected Artifactory details but no details were configured")
+	}
+	url, err := utils.BuildArtifactoryUrl(rtDetails.GetUrl(), "api/system/usage", make(map[string]string))
+	if err != nil {
+		return err
+	}
+	clientDetails := rtDetails.CreateHttpClientDetails()
+	// Check Artifactory version
+	artifactoryVersion, err := rtDetails.GetVersion()
+	if err != nil {
+		return err
+	}
+	if !isCompatible(artifactoryVersion) {
+		return fmt.Errorf("Expected Artifactory version %s or above, got %s", SystemUsageApi, artifactoryVersion)
+	}
+
+	bodyContent, err := reportUsageToJson(productId, commandName)
+	if err != nil {
+		return err
+	}
+	utils.AddHeader("Content-Type", "application/json", &clientDetails.Headers)
+	resp, body, err := serviceManager.Client().SendPost(url, bodyContent, &clientDetails)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return errors.New("Artifactory response: " + resp.Status + "\n" + clientutils.IndentJson(body))
+	}
+
+	log.Debug("Artifactory response:", resp.Status)
+	return nil
+}
+
+// Returns an error if the Artifactory version is not compatible
+func isCompatible(artifactoryVersion string) bool {
```

Comment:
I suggest renaming the function name to *isSupportedVersion*.
If we change the name, let's also rename the test for this function accordingly.

Reply:
Per our discussion, changed it to isVersionCompatible.